pkg/kubelet/cm/topologymanager/policy_single_numa_node.go

```diff
@@ -37,7 +37,7 @@ func (p *singleNumaNodePolicy) Name() string {
 }
 
 func (p *singleNumaNodePolicy) CanAdmitPodResult(hint *TopologyHint) lifecycle.PodAdmitResult {
-	if !hint.Preferred || hint.NUMANodeAffinity.Count() > 1 {
+	if !hint.Preferred {
```

Comment:
I see how this would fix the issue because of the one-off hack we currently have in the `TopologyManager`, i.e. where we intercept hints that have more than one NUMA affinity and force them to `false`.

However, since this is only broken on master, I'm inclined to wait for a proper fix in the form of https://github.com/kubernetes/kubernetes/issues/83478.

What do others think?